### PR TITLE
fix: Fixed diverse typos and incorrect references in scripts

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

Fixed diverse typos and incorrect references in scripts
Fun fact: The branch name has a typo too. You're welcome.

## Pipeline Reference

- Not added as the changes are almost all in failure states 
- The one excemption is the image template removal which is also in a tough spot as its currently not deployable

Just for fun:
[avm.res.analysis-services.server](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.analysis-services.server.yml)

## Type of Change

- [x] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

File: utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourceRemoval.ps1 [hunk 1]
```diff
@@ -156,6 +156,7 @@ function Invoke-ResourceRemoval {
 
             $resourceGroupName = $ResourceId.Split('/')[4]
             $resourceName = Split-Path $ResourceId -Leaf
+            $subscriptionId = $ResourceId.Split('/')[2]
```
Comment (eriqua) [inline utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourceRemoval.ps1:159]: Please add a couple of pipeline badges

Comment (AlexanderSehr) [reply utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourceRemoval.ps1:159]: Just added a badge. However like mentioned in the PRs description none of the pipeline badges could actually touch on the code that was changed

Comment (AlexanderSehr) [reply utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourceRemoval.ps1:159]: Passed

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**